test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -1910,6 +1913,235 @@ public class Class1
             }
         }
 
+        // Test that NuGet packages of the project are added as dependencies 
+        // even if there is already an indirect depenency, provided that the 
+        // project requires a higher version number than the indirect dependency.
+        [Theory]
+        [InlineData("packages.config")]
+        public void PackCommand_PackagesAddedAsDependenciesIfProjectRequiresHigerVersionNumber(string packagesConfigFileName)
+        {
+            var nugetexe = Util.GetNuGetExePath();
+            var workingDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+ 
+            try
+            {
+                Directory.CreateDirectory(workingDirectory);
+                var proj1Directory = Path.Combine(workingDirectory, "proj1");
+                var packagesFolder = Path.Combine(proj1Directory, "packages");
+                Directory.CreateDirectory(proj1Directory);
+                Directory.CreateDirectory(packagesFolder);
+ 
+                // create project 1
+                Util.CreateFile(
+                    proj1Directory,
+                    "proj1.csproj",
+@"<Project ToolsVersion='4.0' DefaultTargets='Build' 
+    xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <OutputPath>out</OutputPath>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include='proj1_file1.cs' />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include='proj1_file2.txt' />
+  </ItemGroup>
+  <Import Project='$(MSBuildToolsPath)\Microsoft.CSharp.targets' />
+</Project>");
+                Util.CreateFile(
+                    proj1Directory,
+                    "proj1_file1.cs",
+@"using System;
+ 
+namespace Proj1
+{
+    public class Class1
+    {
+        public int A { get; set; }
+    }
+}");
+                Util.CreateFile(
+                    proj1Directory,
+                    packagesConfigFileName,
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<packages>
+  <package id=""testPackage1"" version=""1.1.0"" targetFramework=""net45"" />
+  <package id=""testPackage2"" version=""1.2.0"" targetFramework=""net45"" />
+  <package id=""testPackage3"" version=""1.3.0"" targetFramework=""net45"" />
+</packages>");
+                var packageDependencies =
+                    new List<PackageDependencyGroup>()
+                    {
+                        new PackageDependencyGroup(
+                            new NuGetFramework("net45"),
+                            new List<Packaging.Core.PackageDependency>()
+                            {
+                                new Packaging.Core.PackageDependency(
+                                      "testPackage2",
+                                      new VersionRange(new NuGetVersion("1.0.0"), includeMinVersion: true)),
```

Comment:
Add a test that includes a dependency with no range, and one with includeMinVersion: false